app/extensions.js

```diff
@@ -137,19 +151,52 @@ let generateBraveManifest = () => {
     cspDirectives['style-src'] = '\'self\' \'unsafe-inline\' http://' + devServer
   }
 
+  baseManifest.content_security_policy = concatCSP(cspDirectives)
+
+  return baseManifest
+}
+
+// Returns the Chromium extension manifest for the torrentExtension
+// The torrentExtension handles magnet: URLs
+// Analagous to the PDFJS extension, it shows a special UI for that type of resource
+let generateTorrentManifest = () => {
+  let cspDirectives = {
+    'default-src': '\'self\'',
+    'form-action': '\'none\'',
+    'referrer': 'no-referrer',
+    'style-src': '\'self\' \'unsafe-inline\'',
+    'img-src': '* data: blob:'
+  }
+
   // TODO:
-  // * Move WebTorrent to its own renderer process, similar to the way it's done in
-  //   WebTorrent Desktop
+  // * Move WebTorrent to its own process, similar to the way it's done in WebTorrent Desktop
   // * Remove this CSP exception:
   cspDirectives['connect-src'] = '*'
 
-  var csp = ''
-  for (var directive in cspDirectives) {
-    csp += directive + ' ' + cspDirectives[directive] + '; '
+  return {
+    name: 'Torrent Viewer',
+    manifest_version: 2,
+    version: '1.0',
+    content_security_policy: concatCSP(cspDirectives),
+    content_scripts: [],
+    permissions: [
+      'externally_connectable.all_urls', 'tabs', '<all_urls>'
+    ],
+    externally_connectable: {
+      matches: [
+        '<all_urls>'
+      ]
+    },
+    icons: {
+      128: 'img/webtorrent-128.png',
+      48: 'img/webtorrent-48.png',
+      16: 'img/webtorrent-16.png'
+    },
+    web_accessible_resources: [
+      'img/favicon.ico'
+    ],
+    incognito: 'spanning'
```

Comment:
Here's the documentation on the `incognito` property: https://developer.chrome.com/extensions/manifest/incognito

I think `spanning` is fine (as we discussed in person) since when we're done putting webtorrent in it's own process, all the chrome extension will do is 1) serve static html, and 2) relay ipc messages. There's no state (local storage, session, or otherwise) so there should be no way to correlate normal and incognito windows, even with a spanning chrome extension. That said, it wouldn't hurt to change this to `split`. Shouldn't make a difference, so let's go with `split`?

@diracdeltas - does this seem reasonable to you?